src/shutdown.cpp

```diff
@@ -20,6 +21,8 @@
 #include <condition_variable>
 #endif
 
+static std::atomic<int>* g_exit_status{nullptr};
```

Comment:
I am guessing we need a new global here because we can't otherwise access `NodeContext::exit_status`?

Reply:
FWIW, this could be de-globalized again by the interrupt class introduced in https://github.com/bitcoin/bitcoin/pull/27861.